src/woo_publications/publications/formset.py

```diff
@@ -4,15 +4,33 @@
 
 from woo_publications.logging.admin_tools import AuditLogInlineformset
 
-from .tasks import index_document
+from .constants import PublicationStatusOptions
+from .tasks import index_document, remove_document_from_index
 
 
+# TODO: send the download_url with the index_document task.
 class DocumentAuditLogInlineformset(AuditLogInlineformset):
     def save_new(self, form, commit=True):
         document = super().save_new(form, commit)
-        transaction.on_commit(partial(index_document.delay, document_id=document.pk))
+        if document.publicatiestatus == PublicationStatusOptions.published:
+            transaction.on_commit(
+                partial(index_document.delay, document_id=document.pk)
+            )
         return document
 
     def save_existing(self, form, obj, commit=True):
```

Comment:
We either need to find a way to send the download_url with this or remove the functionality of this. Because if you would update a Document without a download_url the Index will overwrite the `attachemnt.content` to `None`.

Reply:
I've resolved this in 5abc7cd55e1e84dc168281a47b14a6b8bef6112c - it's actually possible with built-in django constructs :) The thread-safety issue is not a concern here because the formset instances are always created newly for each request, it's only the `ModelAdmin` instances themselves that have this issue.